clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -117,8 +117,7 @@ public boolean shouldRecord(final int configId) {
     private void checkForest(Set<Sensor> sensors) {
         if (!sensors.add(this))
             throw new IllegalArgumentException("Circular dependency in sensors: " + name() + " is its own parent.");
-        for (int i = 0; i < parents.length; i++)
-            parents[i].checkForest(sensors);
+        for (Sensor parent : parents) parent.checkForest(sensors);
```

Comment:
The convention (even though not enforced by checkstyle) is for `parent.checkForest(sensors);` to go in the next line for the Java code.